training/tensorboard/pytorch/pytorch_smdataparallel_mnist_demo.ipynb

```diff
@@ -0,0 +1,811 @@
+    "\n",
+    "When collecting the model tensors from a distributed training job, using the `SummaryWriter` across multiple ranks can cause a duplication of data collection, so make sure to specify the rank for the `SummaryWriter`. The following code snippet shows the necessary changes to set up the SummaryWriter class and collect model tensors.\n",
+    "\n",
+    "\n",
+    "```\n",
+    "from torch.utils.tensorboard import SummaryWriter\n",
+    "\n",
+    "...\n",
+    "args.rank = rank = dist.get_rank()\n",
+    "args.local_rank = local_rank = int(os.getenv(\"LOCAL_RANK\", -1))\n",
+    "...\n",
+    "writer = SummaryWriter('/opt/ml/output/tensorboard') if rank == 0 and local_rank == 0 else None\n",
+    "...\n",
+    "\n",
+    "def train(args, model, device, train_loader, optimizer, epoch):\n",
+    "    ...\n",
+    "    # Only write on rank 0 to avoid duplicate entries\n",
+    "    if writer:\n",
+    "        writer.add_scalar(\"example_metric\", example_metric)\n",
+    "        \n",
+    "# Close writer at the end of training \n",
+    "writer.close()\n",
+    "```\n",
+    "To see the full training script, run the cell below:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "!pygmentize code/train_pytorch_smdataparallel_mnist.py"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Estimator function options\n",
+    "\n",
+    "To instruct SageMaker to find the local path where the model tensors are saved and upload the tensor data to an Amazon S3 bucket, configure the `sagemaker.debugger.TensorBoardOutputConfig` class as follows.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from sagemaker.debugger import TensorBoardOutputConfig\n",
+    "\n",
+    "tensorboard_s3_output_path = \"s3://{}/{}\".format(bucket, prefix)\n",
+    "\n",
+    "# Create a TensorBoardOutputConfig object to configure automatic data upload from\n",
+    "# the training job's local data storage to S3\n",
+    "tensorboard_output_config = TensorBoardOutputConfig(\n",
+    "    s3_output_path=tensorboard_s3_output_path,\n",
+    "    container_local_output_path=\"/opt/ml/output/tensorboard\",\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from sagemaker.pytorch import PyTorch\n",
+    "\n",
+    "env = {\n",
+    "    \"SAGEMAKER_REQUIREMENTS\": \"requirements.txt\",  # tensorboard dependency required for PyTorch\n",
+    "}\n",
+    "\n",
+    "estimator = PyTorch(\n",
+    "    base_job_name=\"pytorch-tensorboard-dataparallel-mnist\",\n",
+    "    source_dir=\"code\",\n",
+    "    entry_point=\"train_pytorch_smdataparallel_mnist.py\",\n",
+    "    role=role,\n",
+    "    framework_version=\"1.11.0\",\n",
```

Comment:
maybe we want to use 1.13.1, the latest version supported by SageMaker and SMDDP?